### PR TITLE
Update README.md and INSTALL.md for v6.0.0

### PR DESCRIPTION
**Pull request by sttts**
Prepare for v6.0.0 release.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -112,10 +113,11 @@ between client-go versions.
 |----------------|--------------------------------------|-------------------------------|
 | client-go 1.4  | Kubernetes main repo, 1.4 branch     | = -                           |
 | client-go 1.5  | Kubernetes main repo, 1.5 branch     | = -                           |
-| client-go 2.0  | Kubernetes main repo, 1.5 branch     | ✓                             |
-| client-go 3.0  | Kubernetes main repo, 1.6 branch     | ✓                             |
+| client-go 2.0  | Kubernetes main repo, 1.5 branch     | = -                           |
+| client-go 3.0  | Kubernetes main repo, 1.6 branch     | = -                           |
```
Comment (sttts) [inline README.md]: these are 9 months ago. So we should officially deprecate them.

Comment (sttts) [reply README.md]: (we promise 6 months)

Comment (sttts) [reply README.md]: Waiting for approval by @cheftako as both @caesarxuchao and @lavalamp are out.

File: INSTALL.md [hunk 2]
```diff
@@ -99,7 +99,7 @@ your project:
 package: ( your project's import path ) # e.g. github.com/foo/bar
 import:
 - package: k8s.io/client-go
-  version: v2.0.0
```
Comment (cheftako) [inline INSTALL.md]: We clearly need to "fix" v2 but I'm a little worried that we are prescribing client v6. It seems live the "right" version is the fully compatible version for your kubernetes cluster or the latest such version if you have multiple kubernetes clusters with different versions. I am not seeing anything in the docs which explains that or that this the line you would want to change.

Comment (sttts) [reply INSTALL.md]: It's documented in README.md (https://github.com/kubernetes/client-go/pull/347/files#diff-04c6e90faac2675aa89e2176d2eec7d8R82). But your comment shows once again that all this is far too complicated and spread over too many files. We cannot change this here in the PR, but I am fore a major rewrite of the docs here. Compare https://github.com/kubernetes/client-go/pull/337 from your colleague who struggled as well from this documentation.

File: README.md (hunk 1 above)
Comment (sttts) [reply README.md]: @cheftako I read your 👍  as agreement.

**Comment by sttts** [comment]
Merging this, with agreement on https://github.com/kubernetes/client-go/pull/347/files#r157535202.

I think we all agree that we have to work on these two files and simplify things. But we should do that in follow-ups, it's nothing specific to v6.

**Pull request merged**